### PR TITLE
Enable test setup for Mellanox CX4 SR-IOV on ppc64le

### PR DESCRIPTION
**Pull request by bssrikanth**
This patch introduces support for setting up SR-IOV devices
on Mellanox CX4 hardware on ppc64le platform.

Signed-off-by: Srikanth Aithal <sraithal@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by bssrikanth** [comment]
Attaching test results using sr_iov_sanity.py.

**Comment by bssrikanth** [comment]
[job.txt](https://github.com/avocado-framework/avocado-vt/files/582953/job.txt)

**Comment by bssrikanth** [comment]
@ldoktor not sure why seeing travisCI errors..

**Comment by ldoktor** [comment]
Hello @bssrikanth, the pep8 was updated in Travis being more pedantic and causing lots of style issues. It was already fixed so I re-triggered the CI and now all the failures should be related to this PR.

**Comment by bssrikanth** [comment]
thank you @ldoktor  I am working on fixing the failure reported

File: virttest/test_setup.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 import random
 import math
 import shutil
+import arch
+from netaddr import *
```
Comment (xutian) [inline virttest/test_setup.py]: Please update requirements.txt  too

Comment (bssrikanth) [inline virttest/test_setup.py]: @xutian I have updated requirements.txt.. same prereq package installation is going through fine on my x86 laptop, but seeing travis-ci reporting issues :-(

Comment (bssrikanth) [inline virttest/test_setup.py]: I have removed arch dependencies and have used platform

**Comment by bssrikanth** [comment]
I have no removed arch dependecies in test_setup.py. This revised version is tested and the job.txt test output holds good.

Please let me know your review comments

**Comment by bssrikanth** [comment]
@xutian @will-Do @PandaWei please let me know if you have any feedback here. Many thanks in advance :-)

File: virttest/test_setup.py [hunk 2]
```diff
@@ -1145,11 +1156,19 @@ def get_pf_vf_info(self):
                 vf_ids.append(vf_info)
             pf_info["vf_ids"] = vf_ids
             pf_vf_dict.append(pf_info)
+        # When the VFs loaded on a PF are > 10 [I have tested till 63 which is
+        # max VF supported by Mellanox CX4 cards],VFs probe on host takes bit
+        # more time than usual.
+        # Without this sleep ifconfig would fail With below error, since not all
+        # probed interfaces of VF would have got properly renamed:
+        # CmdError: Command 'ifconfig -a' failed (rc=1)
+        if (int(self.driver_option) > 10):
```
Comment (xutian) [inline virttest/test_setup.py]: useless '()'

**Comment by xutian** [comment]
Looks OK to me.

**Comment by bssrikanth** [comment]
@xutian thank you!. I have addressed your comment.

**Comment by xutian** [comment]
Thank you for @bssrikanth  contribution， merged.

**Pull request merged**